app/src/main/java/co/csadev/fetchhiring/ui/ItemScreen.kt

```diff
@@ -0,0 +1,110 @@
+package co.csadev.fetchhiring.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import co.csadev.fetchhiring.R
+import co.csadev.fetchhiring.data.Item
+import co.csadev.fetchhiring.data.ItemState
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.SwipeRefreshState
+
+/**
+ * The Item Screen layout.
+ *
+ * @param itemsState UI state for the Items
+ * @param refreshState Stateful value to manage showing refresh UI
+ * @param modifier the modifier to apply to this layout.
+ * @param onRefresh callback to trigger a refresh
+ */
+@Composable
+fun ItemScreen(
+    itemsState: ItemState,
+    refreshState: SwipeRefreshState,
+    modifier: Modifier = Modifier,
+    onRefresh: () -> Unit
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(colorResource(id = R.color.surface))
+    ) {
+        SwipeRefresh(
+            state = refreshState,
+            onRefresh = onRefresh,
+            modifier = modifier
+                .fillMaxSize()
+                .background(colorResource(id = R.color.surface))
+        ) {
+            if (refreshState.isRefreshing) {
```

Comment:
This shows 2 progress circles when initiated. The one at the top is the default and should either be used or disabled.

Reply:
Great improvement!